students/shodges/lesson09/donor_models.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+import shelve
+from collections import OrderedDict
+from pathlib import Path
+from datetime import datetime
+
+class Donor(object):
+    """
+    Base individual donor class.
+
+    donor_name is the only required argument; all additional arguments will be converted to
+    floats and added as initial donations.
+
+    Each instance has two variables:
+      * name -- the donor's name
+      * ledger -- a list with all donations as floats
+    """
+    def __init__(self, donor_name, *args):
+        self.name = donor_name
+        self.ledger = []
+        for i in args:
+            self.ledger.append(float(i))
+
+
+    @property
+    def count(self):
+        """
+        Returns the number of donations in the donor's ledger.
+        """
+        return len(self.ledger)
+
+
+    @property
+    def donations(self):
+        """
+        Returns the total of all donations in the donor's ledger.
+        """
+        return sum(self.ledger)
+
+
+    def process(self, donation):
+        """
+        Appends donation to the donor's ledger.  Returns True if successful.  No exceptions
+        raised are caught and are required to be caught by the implementing code.
+        """
+        self.ledger.append(float(donation))
+        return True
```

Comment:
why return True? not needed here

Reply:
You're right.  For some things I had been checking the return but not here (although I did in a few unit tests so those are fixed).